server.js

```diff
@@ -1,33 +1,98 @@
-import express from 'express'
-import bodyParser from 'body-parser'
-import cors from 'cors'
-
-// If you're using one of our datasets, uncomment the appropriate import below
-// to get started!
-// 
-// import goldenGlobesData from './data/golden-globes.json'
-// import avocadoSalesData from './data/avocado-sales.json'
-// import booksData from './data/books.json'
-// import netflixData from './data/netflix-titles.json'
-// import topMusicData from './data/top-music.json'
-
-// Defines the port the app will run on. Defaults to 8080, but can be 
+import express from "express";
+import bodyParser from "body-parser";
+import cors from "cors";
+
+import booksData from "./data/books.json";
+
+const ERROR_NOT_FOUND = { error: "No book results were found" };
+
+// Defines the port the app will run on. Defaults to 8080, but can be
 // overridden when starting the server. For example:
 //
 //   PORT=9000 npm start
-const port = process.env.PORT || 8080
-const app = express()
+const port = process.env.PORT || 8080;
+const app = express();
+const listEndpoints = require("express-list-endpoints");
 
-// Add middlewares to enable cors and json body parsing
-app.use(cors())
-app.use(bodyParser.json())
+// Add middlewares to enable cors and json body parsing - makes it easier for frontend to use the API. Corrs allows API's to say where requests can come from. To add extra security
+app.use(cors());
+app.use(bodyParser.json()); //This allows Express to read JSON in POST requests
 
 // Start defining your routes here
-app.get('/', (req, res) => {
-  res.send('Hello world')
-})
+app.get("/", (req, res) => {
+  res.send(listEndpoints(app));
+});
+
+app.get("/books", (req, res) => {
+  let page = req.query.page ?? 1;
+  const pageSize = req.query.pageSize ?? 20; // this allows you to query page numbers, example: http://localhost:8080/books?page=1&pageSize=10
+  page = +page - 1;
+  let bookList = booksData;
+  console.log(`page = ${page}`);
+  const startIndex = page * pageSize; //calculate the start index
+  const endIndex = startIndex + +pageSize;
+  const booksForPage = bookList.slice(startIndex, endIndex); //Slice creates a new array to limit the number of books returned to 20 per page
+
+  if (bookList.length === 0) {
+    res.status(404).send(ERROR_NOT_FOUND);
+  }
+  const returnObject = {
+    pageSize: booksForPage.length,
+    currentPage: page + 1,
+    maxPages: Math.ceil(bookList.length / pageSize),
+    totalNumOfBooks: bookList.length,
+    results: booksForPage,
+  };
+  res.send(returnObject);
+});
+
+app.get("/books/bookauthor", (req, res) => {
```

Comment:
It would be more restfull to create this endpoint with normal parameter :

```
app.get("/books/bookauthor/:author", (req res) => { ... });
```

Think about query parameteres as about something optional - we can provide it in URL, but we don't have to.
Since this endpoint is created for book authors only, it's reasonable to build it with normal parameter, like suggested above